data-prepper-plugins/rds-source/src/main/java/org/opensearch/dataprepper/plugins/source/rds/datatype/impl/SpatialTypeHandler.java

```diff
@@ -4,22 +4,364 @@
 import org.opensearch.dataprepper.plugins.source.rds.model.TableMetadata;
 import org.opensearch.dataprepper.plugins.source.rds.datatype.MySQLDataType;
 
+import java.nio.ByteBuffer;
+import java.nio.ByteOrder;
+import java.util.ArrayList;
+import java.util.List;
 import java.util.Map;
 
+/**
+ * Handles spatial data types for OpenSearch geo_shape field type compatibility.
+ * This handler supports various geometric shapes including Point, LineString, Polygon,
+ * MultiPoint, MultiLineString, MultiPolygon, and GeometryCollection.
+ *
+ * @see <a href="https://opensearch.org/docs/latest/field-types/supported-field-types/geo-shape/">OpenSearch Geo Shape Documentation</a>
+
+ * Supported WKT (Well-Known Text) formats:
+ * <ul>
+ *   <li>POINT(x y)</li>
+ *   <li>LINESTRING(x y, x y, ...)</li>
+ *   <li>POLYGON((x y, x y, ..., x y))</li>
+ *   <li>MULTIPOINT((x y), (x y), ...)</li>
+ *   <li>MULTILINESTRING((x y, x y, ...), (x y, x y, ...))</li>
+ *   <li>MULTIPOLYGON(((x y, x y, ...)), ((x y, x y, ...)))</li>
+ *   <li>GEOMETRYCOLLECTION(POINT(x y), LINESTRING(x y, x y))</li>
+ * </ul>
+ * */
 public class SpatialTypeHandler implements DataTypeHandler {
+    // MySQL geometry type constants
+    private static final int GEOMETRY_POINT = 1;
+    private static final int GEOMETRY_LINESTRING = 2;
+    private static final int GEOMETRY_POLYGON = 3;
+    private static final int GEOMETRY_MULTIPOINT = 4;
+    private static final int GEOMETRY_MULTILINESTRING = 5;
+    private static final int GEOMETRY_MULTIPOLYGON= 6;
+    private static final int GEOMETRY_COLLECTION = 7;
```

Comment:
Are we missing a Geometry type here? It's the singular form of GeometryCollection type

Reply:
Yes, Geometry type is handled under GeometryCollection type